src/lib/mina_graphql/mina_graphql.ml

```diff
@@ -4283,21 +4283,22 @@ module Queries = struct
       ~resolve:(fun { ctx = mina; _ } () token_id ->
         match get_ledger_and_breadcrumb mina with
         | Some (ledger, breadcrumb) ->
-            let token_accounts =
-              Ledger.accounts ledger
-              |> Account_id.Set.filter ~f:(fun acct_id ->
-                     Token_id.equal (Account_id.token_id acct_id) token_id )
-              |> Account_id.Set.to_list
-            in
-            List.map token_accounts ~f:(fun acct_id ->
-                (* we found these account ids in the ledger, lookup should always succeed *)
-                let loc =
-                  Option.value_exn @@ Ledger.location_of_account ledger acct_id
-                in
-                let account = Option.value_exn @@ Ledger.get ledger loc in
-                Types.AccountObj.Partial_account.of_full_account ~breadcrumb
-                  account
-                |> Types.AccountObj.lift mina account.public_key )
+            let accounts = Ledger.accounts ledger in
+            Account_id.Set.fold accounts ~init:[] ~f:(fun acct_objs acct_id ->
+                if Token_id.equal (Account_id.token_id acct_id) token_id then
```

Comment:
Nit: how do you feel about inverting the conditional like so:
 `if Token_id.((Account_id.token_id acct_id) <> token_id) then acct_objs else ....`

I feel it is somewhat easier to read. Feel free to resolve either way.